### ISSUE TITLE
[dojoup] ensure the script fails for invalid version

### ISSUE THREAD

**Issue by glihm** [dojoengine/dojo#1831]
**Describe the bug**
Currently, `dojoup` scripts can be used to install a specific version using `dojoup -v ...`. However, if an invalid version is passed to the script, it does not fail.

**To Reproduce**
```bash
# Install dojoup
curl -L https://install.dojoengine.org | bash

# Run dojo up with invalid version
dojoup -v 9237.4245.5
```

**Expected behavior**
The program should stop with a comprehensive error message.

**Additional context**
The script is written into the [dojo repository](https://github.com/dojoengine/dojo/tree/0227d3759961acf0c71ed4bd077c53ad9e5d454d/dojoup).

**Comment by bhavyagosai** [comment]
I assume that we would fetch a list of release versions for dojo using the GitHub API to check it with the mentioned release version if it exists or not? Or does there seem to be an easier way?

 I seem eager to work on this. If it is up for grabs, can you assign it to me? @glihm

**Comment by glihm** [comment]
> I assume that we would fetch a list of release versions for dojo using the GitHub API to check it with the mentioned release version if it exists or not? Or does there seem to be an easier way?
> 
> I seem eager to work on this. If it is up for grabs, can you assign it to me? @glihm

Yup! You may fetch a list of releases, or I guess checking the `curl` return value may also be an indicator of the release validity.

Assigned, thank you for contributing!

**Comment by bhavyagosai** [comment]
I have noticed something btw

https://github.com/dojoengine/dojo/blob/89a5b9eb31a32e58a0429b7d17151361cb09f746/dojoup/dojoup#L83-L91

Here, as `per_page` query parameter has not been mentioned in the api endpoint, it defaults to `30` ([according to github api](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases)) and only first 30 release tags are fetched. This currently doesn't affect us, as this piece of code is being used to fetch the _**latest**_ non-prerelease version number, excluding alpha releases. But still just wanted to mention it here, if anything unwanted was happening.

Rest assured, I have raised a PR for the issue mentioned. Hope it helps :)
